homeassistant/components/wallbox/__init__.py

```diff
@@ -47,7 +47,7 @@ def _authenticate(self):
             return True
         except requests.exceptions.HTTPError as wallbox_connection_error:
             if wallbox_connection_error.response.status_code == 403:
-                raise InvalidAuth from wallbox_connection_error
+                raise exceptions.Unauthorized from wallbox_connection_error
```

Comment:
`Unauthorized` should be used when something isn't authorized according to the Home Assistant user authorization system. We shouldn't reuse that exception here.

I'm not sure what the feedback was around this in the original PR, but we normally either use the integration library exceptions or define custom exceptions in the integration that inherit `HomeAssistantError`.

Finally, regardless of what we decide here, I suggest we don't change this part in the same PR as we refactor the tests. Focus on the test refactor in this PR and doa separate PR around exceptions if needed.

Reply:
It's ok to create a custom exception in init module, but I suggest we do those changes in a separate PR so we keep this PR focused on the tests. They shouldn't be affected by that change.